File: app/src/main/java/ru/p3tr0vich/fuel/FragmentDialogMessage.java
package ru.p3tr0vich.fuel;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.support.v4.app.FragmentActivity;
import android.support.v7.app.AlertDialog;
import android.text.TextUtils;
import android.widget.TextView;

public class FragmentDialogMessage extends DialogFragment {

    private static final String TAG = "FragmentDialogMessage";

    private static final String TITLE = "title";
    private static final String MESSAGE = "message";

    private static FragmentDialogMessage newInstance(@Nullable String title, @NonNull String message) {
        Bundle args = new Bundle();
        args.putString(TITLE, title);
        args.putString(MESSAGE, message);

        FragmentDialogMessage dialogMessage = new FragmentDialogMessage();
        dialogMessage.setArguments(args);

        return dialogMessage;
    }

    public static void show(@NonNull FragmentActivity parent,
                            @Nullable String title,
                            @NonNull String message) {
        newInstance(title, message).show(parent.getSupportFragmentManager(), TAG);
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        Bundle arguments = getArguments();

        assert arguments != null;

        String title = arguments.getString(TITLE);

        Activity activity = getActivity();

        assert activity != null;

        AlertDialog.Builder builder = new AlertDialog.Builder(activity);

        if (!TextUtils.isEmpty(title)) {
            @SuppressLint("InflateParams")
            TextView customTitle = (TextView) activity.getLayoutInflater()
                    .inflate(R.layout.apptheme_dialog_title, null, false);

            customTitle.setText(title);

            builder.setCustomTitle(customTitle);
        }

        builder.setMessage(arguments.getString(MESSAGE));

        builder.setPositiveButton(R.string.dialog_btn_ok, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int id) {
                dismiss();
            }
        });

        return builder.setCancelable(true).create();
    }
}

File: app/src/main/kotlin/ru/p3tr0vich/fuel/fragments/FragmentDialogMessage.kt
package ru.p3tr0vich.fuel.fragments

import android.annotation.SuppressLint
import android.app.Dialog
import android.os.Bundle
import android.text.TextUtils
import android.widget.TextView
import androidx.annotation.StringRes
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.FragmentActivity
import ru.p3tr0vich.fuel.R

class FragmentDialogMessage : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(requireActivity())

        val title = arguments?.getString(TITLE)

        if (!TextUtils.isEmpty(title)) {
            @SuppressLint("InflateParams")
            val customTitle = requireActivity().layoutInflater
                    .inflate(R.layout.apptheme_dialog_title, null, false) as TextView

            customTitle.text = title

            builder.setCustomTitle(customTitle)
        }

        builder.setMessage(arguments?.getString(MESSAGE))

        builder.setPositiveButton(R.string.dialog_btn_ok) { _, _ -> dismiss() }

        return builder.setCancelable(true).create()
    }

    companion object {

        private const val TAG = "FragmentDialogMessage"

        private const val TITLE = "title"
        private const val MESSAGE = "message"

        private fun newInstance(title: String?, message: String): FragmentDialogMessage {
            val args = Bundle()
            args.putString(TITLE, title)
            args.putString(MESSAGE, message)

            val dialogMessage = FragmentDialogMessage()
            dialogMessage.arguments = args

            return dialogMessage
        }

        fun show(parent: FragmentActivity, title: String?, message: String) {
            newInstance(title, message).show(parent.supportFragmentManager, TAG)
        }

        fun show(parent: FragmentActivity, @StringRes titleId: Int?, @StringRes messageId: Int) {
            show(parent, if (titleId == null) null else parent.getString(titleId), parent.getString(messageId))
        }
    }
}